Modernising the entire project

bradrushworth opened the pull request:
sdk: '>=2.18.4 <3.0.0'
added new device "Pixel 4" #195
Fix a typo in the documentation #201
Corrects the iPhone 12 screen size and safe area #204
Bump intl to v.0.18.0 #222

wwkler committed:
iphone 12 ratio changed

Uuttssaavv committed:
changed the screen size for iphone 12

Uuttssaavv committed:
added custom device pixel 4

Uuttssaavv committed:
added pixel ratio for device pixel 4

walid-ashik committed:
Fix a typo

walid-ashik committed:
Fix a typo in the documentation

walid-ashik committed:
Wip: Add iPhone 14 Pro Max device support

heeyunlee committed:
Revert "fill in 4th gen stats"

This reverts commit 0c1b94a4f8120955df3ce52b1bb9778a282ca586.

heeyunlee committed:
update changelog and version for device_frame package

bradrushworth committed:
sdk: '>=2.18.4 <3.0.0'

bradrushworth committed:
Merge remote-tracking branch 'wwkler/develop'

bradrushworth committed:
Merge remote-tracking branch 'Uuttssaavv/issue136'

bradrushworth committed:
Merge remote-tracking branch 'walid-ashik/patch-1'

bradrushworth committed:
Merge remote-tracking branch 'walid-ashik/patch-2'

bradrushworth committed:
sdk: '>=2.18.4 <3.0.0'

bradrushworth committed:
Merge remote-tracking branch 'heeyunlee/master'

bradrushworth committed:
Cleaning up Pixel 4 implementation.

bradrushworth committed:
Revert "Wip: Add iPhone 14 Pro Max device support"

This reverts commit 708c6b4b273fee57d3fa544c8e22a769895529d2.

bradrushworth committed:
More sensible spot for Pixel 4.

bradrushworth commented:
Feel free to use this update by adding to your pubspec.yaml

```
dependency_overrides:
  device_preview:
    git:
      url: https://github.com/bradrushworth/flutter_device_preview.git
      path: device_preview
#    path: ../flutter_device_preview/device_preview
  device_frame:
    git:
      url: https://github.com/bradrushworth/flutter_device_preview.git
      path: device_frame
#    path: ../flutter_device_preview/device_frame
```

bradrushworth committed:
Revert "added custom device pixel 4"

This reverts commit e1ef7ebd

bradrushworth committed:
Revert "Revert "added custom device pixel 4""

This reverts commit 337578fbd47011178bbfef8c81492d47874cb86e.

mikhael28 commented:
This is a sweet update! I hope the author merges it in

bradrushworth committed:
Revert "Revert "Wip: Add iPhone 14 Pro Max device support""

This reverts commit 8db292edb620a1b6471a5e57f01f70f77b8264a5.

bradrushworth committed:
sdk: ">=2.19.6 <3.0.0"

bradrushworth committed:
Fixed NaN entries from https://fluttershapemaker.com/ for iPhone 14 Pro Max device

wer-mathurin commented:
@aloisdeniel: this plugin is awesome, but need some love in terms of accepting PR.  PRs that make it compatible with the latest and greatest flutter release and devices seems very helpful.  Any timeframe to review this PR?

Princewil commented:
> Feel free to use this update by adding to your pubspec.yaml
> 
> ```
> dependency_overrides:
>   device_preview:
>     git:
>       url: https://github.com/bradrushworth/flutter_device_preview.git
>       path: device_preview
> #    path: ../flutter_device_preview/device_preview
>   device_frame:
>     git:
>       url: https://github.com/bradrushworth/flutter_device_preview.git
>       path: device_frame
> #    path: ../flutter_device_preview/device_frame
> ```

Sorry, I don't get this. I can see two packages here: "device_preview" and "device_frame", you mean I should add both?

Princewil commented:
> > Feel free to use this update by adding to your pubspec.yaml
> > ```
> > dependency_overrides:
> >   device_preview:
> >     git:
> >       url: https://github.com/bradrushworth/flutter_device_preview.git
> >       path: device_preview
> > #    path: ../flutter_device_preview/device_preview
> >   device_frame:
> >     git:
> >       url: https://github.com/bradrushworth/flutter_device_preview.git
> >       path: device_frame
> > #    path: ../flutter_device_preview/device_frame
> > ```
> 
> Sorry, I don't get this. I can see two packages here: "device_preview" and "device_frame", you mean I should add both?

Never mind. I got it now.
Thanks for the update.

Dnathan33 commented:
Thanks for making the updates @bradrushworth. Good work!

@aloisdeniel I'd be happy to help in maintaining the project. I understand that it can be difficult for one person to maintain it all consistently. I'm already familiar with the project's codebase and I have two PRs already merged in.

walid-ashik commented:
> Thanks for making the updates @bradrushworth. Good work!
> 
> @aloisdeniel I'd be happy to help in maintaining the project. I understand that it can be difficult for one person to maintain it all consistently. I'm already familiar with the project's codebase and I have two PRs already merged in.

There's no activity in this project. can you fork it and create a separate project where we can all contribute? seems like this project is dead! :(

yassine-bennkhay commented:
@bradrushworth Any idea on this issue?
[Dialog widget is out of phone boundaries #175](https://github.com/aloisdeniel/flutter_device_preview/issues/175)

Nikzed commented:
iPhone14Pro is not working here. Do not waste your time

bradrushworth committed:
Upgrade to flutter 2.19

bradrushworth committed:
Upgrade SDK version in pubspec.yaml files

This commit updates the SDK constraints and several dependencies in the `pubspec.yaml` files for `device_frame`, `device_preview`, `devtools_device_preview`, and their respective example projects.

Specifically, the SDK constraint has been updated from `'>=2.19.6 <3.0.0'` to `'>=3.9.2 <4.0.0'` or a similar higher version across the projects.

Dependencies like `freezed_annotation`, `freezed`, and `flutter_lints` have also been upgraded to newer versions.

aloisdeniel commented:
Thank you for this, and I am sorry it sat here so long.

device_preview **3.0.0** is out and it is a from-scratch rebuild: the simulation moved into a custom `WidgetsBinding` below the widget layer, and the in-app toolbar moved into a **Flutter DevTools extension**. The files this PR changes no longer exist on `master`, so there is no way to merge it.

The project has now been modernised the hard way — rewritten.

Closing for that reason — not because the change was wrong. If you would like to bring the idea to 3.0, a fresh PR against the current tree is very welcome.

- Changelog: https://pub.dev/packages/device_preview/changelog
- Migrating from 2.x: https://github.com/aloisdeniel/flutter_device_preview/blob/master/device_preview/README.md#migrating-from-2x
